server/neptune/workflows/internal/activities/db.go

```diff
@@ -3,27 +3,48 @@ package activities
 import (
 	"context"
 
-	"github.com/runatlantis/atlantis/server/neptune/logger"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deployment"
 )
 
+type store interface {
+	GetDeploymentInfo(ctx context.Context, repoName string, rootName string) (*deployment.Info, error)
+	SetDeploymentInfo(ctx context.Context, deploymentInfo deployment.Info) error
+}
+
 type dbActivities struct {
+	DeploymentInfoStore store
 }
 
 type FetchLatestDeploymentRequest struct {
-	RepositoryURL string
-	RootName      string
+	FullRepositoryName string
+	RootName           string
 }
 
 type FetchLatestDeploymentResponse struct {
-	ID       string
-	Revision string
+	DeploymentInfo deployment.Info
 }
 
 func (a *dbActivities) FetchLatestDeployment(ctx context.Context, request FetchLatestDeploymentRequest) (FetchLatestDeploymentResponse, error) {
-	logger.Info(ctx, "fetching latest deployment")
+	deploymentInfo, err := a.DeploymentInfoStore.GetDeploymentInfo(ctx, request.FullRepositoryName, request.RootName)
+	if err != nil {
+		return FetchLatestDeploymentResponse{}, errors.Wrapf(err, "fetching deployment info for %s/%s", request.FullRepositoryName, request.RootName)
+	}
 
 	return FetchLatestDeploymentResponse{
-		ID:       "test-id",
-		Revision: "1234",
+		DeploymentInfo: *deploymentInfo,
```

Comment:
Why are you always dereferencing pointers. If something returns a pointer, you likely should be passing it through no?